### PR TITLE
40.07-40.09 osx mem layouts, correct typo.

### PR DESCRIPTION
**Pull request by fricy**

### CODE REVIEW

File: etc/memory_layouts/osx/v0.40.07_osx.ini [hunk 1]
```diff
@@ -0,0 +1,392 @@
...
+[job_details]
+id=0x0008
+mat_type=0x0024
+mat_index=0x0028
+mat_category=0x003c
+on_break_flag=0x0011
+sub_job_id=0x0040
+reaction=0x0004
+reaction_skill=0x0028
+mat_type=0x0024
```
Comment (splintermind) [inline etc/memory_layouts/osx/v0.40.07_osx.ini]: Last 3 are duplicates.

File: etc/memory_layouts/osx/v0.40.08_osx.ini [hunk 2]
```diff
@@ -0,0 +1,378 @@
...
+
+[job_details]
+id=0x0008
+mat_type=0x0024
+mat_index=0x0028
+mat_category=0x003c
+on_break_flag=0x0011
+sub_job_id=0x0040
+reaction=0x0004
+reaction_skill=0x0028
```
Comment (splintermind) [inline etc/memory_layouts/osx/v0.40.08_osx.ini:300]: Last 3 are duplicates.

File: etc/memory_layouts/osx/v0.40.07_osx.ini [hunk 3]
```diff
@@ -0,0 +1,392 @@
...
+squad_vector=0x0198e3d0
+current_year=0x0187aa40
+cur_year_tick=0x0187aa48
+dwarf_civ_index=0x0187cf68
+races_vector=0x01992c98
+reactions_vector=0x01997220
+historical_figures=0x01998724
+fake_identities=0x0198f838
+historical_figures_vector=0x01998724
+fake_identities_vector=0x0198f838
```
Comment (splintermind) [inline etc/memory_layouts/osx/v0.40.07_osx.ini:19]: fake_identities can be removed (duplicate of fake_identities_vector)

File: etc/memory_layouts/osx/v0.40.08_osx.ini [hunk 4]
```diff
@@ -0,0 +1,378 @@
...
+active_creature_vector=0x01899318
+dwarf_race_index=0x0187cf74
+squad_vector=0x0198e3d0
+current_year=0x0187aa40
+cur_year_tick=0x0187aa48
+dwarf_civ_index=0x0187cf68
+races_vector=0x01992c98
+reactions_vector=0x01997220
+historical_figures=0x01998724
+fake_identities=0x0198f838
```
Comment (splintermind) [inline etc/memory_layouts/osx/v0.40.08_osx.ini]: fake_identities can be removed (duplicate of fake_identities_vector)

File: etc/memory_layouts/osx/v0.40.08_osx.ini [hunk 5]
```diff
@@ -0,0 +1,378 @@
...
+shield_vector=0x01992e1c
+helm_vector=0x01992e28
+pant_vector=0x01992e34
+food_vector=0x01992e40
+colors_vector=0x019971fc
+shapes_vector=0x01997208
+base_materials=0x01997794
+inorganics_vector=0x01992be4
+plants_vector=0x01992bfc
+material_templates_vector=0x01992bd8
```
Comment (splintermind) [inline etc/memory_layouts/osx/v0.40.08_osx.ini:41]: Missing all the itemdef vector offsets.

**Comment by fricy** [comment]
regenned 07-08 with the new script.

File: etc/memory_layouts/osx/v0.40.07_osx.ini [hunk 6]
```diff
@@ -0,0 +1,359 @@
...
+child_name_plural=0x001c
+pref_string_vector=0x0048
+castes_vector=0x0060
+pop_ratio_vector=0x006c
+materials_vector=0x1cf4
+flags=0x0078
+tissues_vector=0x1d00
+
+[caste_offsets]
+caste_name=0x0004
```
Comment (splintermind) [inline etc/memory_layouts/osx/v0.40.07_osx.ini:99]: Missing caste baby/child age.

File: etc/memory_layouts/osx/v0.40.07_osx.ini [hunk 7]
```diff
@@ -0,0 +1,359 @@
...
+caste_att_rates=0x0d60
+caste_att_caps=0x0e90
+adult_size=0x0104
+flags=0x02d4
+extracts=0x13a4
+skill_rates=0x03fc
+body_info=0x02e0
+
+[hist_entity_offsets]
+squads=0x0884
```
Comment (splintermind) [inline etc/memory_layouts/osx/v0.40.07_osx.ini:116]: Missing beliefs.

File: etc/memory_layouts/osx/v0.40.07_osx.ini [hunk 8]
```diff
@@ -0,0 +1,359 @@
...
+id=0x00a8
+animal_type=0x00b0
+civ=0x00b8
+specific_refs=0x00f4
+squad_id=0x010c
+squad_position=0x0110
+recheck_equipment=0x015c
+mood=0x01e4
+birth_year=0x0204
+birth_time=0x0208
```
Comment (splintermind) [inline etc/memory_layouts/osx/v0.40.07_osx.ini:227]: Missing pet_owner_id.

File: etc/memory_layouts/osx/v0.40.07_osx.ini [hunk 9]
```diff
@@ -0,0 +1,359 @@
...
+wounds_vector=0x0358
+mood_skill=0x02ec
+used_items_vector=0x0818
+affection_level=0x000c
+inventory=0x027c
+inventory_item_mode=0x0004
+inventory_item_bodypart=0x0006
+
+[syndrome_offsets]
+cie_effects=0x0004
```
Comment (splintermind) [inline etc/memory_layouts/osx/v0.40.07_osx.ini:260]: Missing cie_end.

**Comment by splintermind** [comment]
[I've just updated the lua script to include the last set of missing offsets.](https://raw.githubusercontent.com/splintermind/Dwarf-Therapist/9e6dc19ef3e78e11607c17368eeda7ec0a969dfc/export-dt-ini.lua)

**Pull request closed without merging**